### PR TITLE
Bug 1219630 - Adding test that checks a search through the rocketbar r=Johan

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6825fcfa81718cf6465a615bc24711b799f288a7

File: apps/search/test/marionette/lib/search.js [hunk 1]
```diff
@@ -51,7 +51,8 @@ Search.prototype = {
    * Navigates to the search results frame.
    */
   goToResults: function() {
-    var iframe = this.client.findElement(Search.Selectors.iframe);
+    var frame = Search.Selectors.iframe;
+    var iframe = this.client.helper.waitForElement(frame);
```
Comment (JohanLorenzo) [inline apps/search/test/marionette/lib/search.js]: What's the advantage of splitting the line into 2?

File: apps/search/test/marionette/search_rocketbar_test.js [hunk 2]
```diff
@@ -100,8 +110,18 @@ marionette('Search - Rocketbar Test', function() {
...
-    rocketbar.switchToBrowserFrame(server.url('sample.html'));
+    rocketbar.enterText(textToSearch, true);
+    rocketbar.switchToBrowserFrame(searchTemplate);
+  });
+
+  test('Browser search', function() {
+    home.waitForLaunch();
+    rocketbar.homescreenFocus();
+    var textToSearch = 'a test';
+    rocketbar.enterText(textToSearch, true);
```
Comment (JohanLorenzo) [inline apps/search/test/marionette/search_rocketbar_test.js]: We could make the test closer to the user's intent if `rocketbar` had a method like:
```js
performSearchInBrowser(textToSearch):
    rocketbar.enterText(textToSearch, true);
    rocketbar.waitForBrowserFrame();
    rocketbar.appTitleFocus();
```

File: apps/system/test/marionette/lib/rocketbar.js [hunk 3]
```diff
@@ -154,19 +154,12 @@ Rocketbar.prototype = {
...
-    this.client.waitFor((function() {
-      var size = this.client.findElement(this.selectors.activeBrowserFrame)
-        .size();
-      return size.width === 320 && size.height === 456;
-    }).bind(this));
-    return this.client.executeScript(function() {
-      window.wrappedJSObject.dispatchEvent(new CustomEvent('home'));
-    });
+    var browser = this.selectors.activeBrowserFrame;
+    this.client.helper.waitForElement(browser);
```
Comment (JohanLorenzo) [inline apps/system/test/marionette/lib/rocketbar.js]: Nice! :+1:

File: apps/search/test/marionette/lib/search.js (hunk 1 above)
Comment (albertopq) [inline apps/search/test/marionette/lib/search.js]: Just avoiding long line. I'll change the name, good idea!

File: apps/search/test/marionette/search_rocketbar_test.js (hunk 2 above)
Comment (albertopq) [inline apps/search/test/marionette/search_rocketbar_test.js]: Mhm, I don't think we always want to switch to the browser frame, but +1 to rephrase this to show user's intent.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e4082928115949021cad931bb6fa6748cccd9693
